packages/@aws-cdk/aws-codeguruprofiler/test/profiling-group.test.ts

```diff
@@ -169,6 +169,13 @@ describe('profiling group', () => {
     });
   });
 
+  test('use name specified via fromProfilingGroupName', () => {
+    const stack = new Stack();
+
+    const profilingGroup = ProfilingGroup.fromProfilingGroupName(stack, 'MyProfilingGroup', 'MyAwesomeProfilingGroup');
+    expect(profilingGroup.profilingGroupName).toEqual('MyAwesomeProfilingGroup');
```

Comment:
Given that it looks right here, I feel pretty confident about your change, I'd just like to see the correct raw output in the integ tests like here.

Reply:
I'm not sure how to trigger the raw output as you requested in the integration tests, which is why I just went with unit tests originally. The problem is forcing the resolution the name instead of the `Ref` token as you noted. I couldn't find any hints of this happening. Any code tips? I thought for this specific fix the unit test sufficed.

I was happy to be able to use test-driven development to both replicate the bug and assert its correctness. I found no value from the integration tests given that I have to accept the snapshot, and as you've pointed out it doesn't actually "cover" the fix.